source/MRViewer/MRSubfeaturePickerWidget.cpp

```diff
@@ -0,0 +1,657 @@
+            subfeature.object->setLineWidth( pickerLineWidth );
+        }
+    }
+    // Add polygon offset to meshes.
+    auto lambda = [&]( auto& lambda, const std::shared_ptr<Object>& cur ) -> void
+    {
+        if ( auto meshHolder = std::dynamic_pointer_cast<ObjectMeshHolder>( cur ) )
+        {
+            auto polygonOffsetMask = meshHolder->getVisualizePropertyMask( MeshVisualizePropertyType::PolygonOffsetFromCamera );
+            if ( !polygonOffsetMask.contains( viewportId ) )
+            {
+                meshRollbackEntries.push_back( {
+                    .meshHolder = meshHolder,
+                    .polygonOffsetMask = polygonOffsetMask,
+                } );
+                meshHolder->setVisualizePropertyMask( MeshVisualizePropertyType::PolygonOffsetFromCamera, ViewportMask::all() );
+            }
+        }
+        for ( const auto& child : cur->children() )
+            lambda( lambda, child );
+    };
+    lambda( lambda, SceneRoot::getSharedPtr() );
+
+    func();
+
+    for ( FeatureRollbackEntry& rollback : featureRollbackEntries )
+    {
+        rollback.subfeature->setVisibilityMask( rollback.visMask );
+        rollback.subfeature->setPointSize( rollback.pointSize );
+        rollback.subfeature->setLineWidth( rollback.lineWidth );
+    }
+    for ( MeshRollbackEntry& rollback : meshRollbackEntries )
+    {
+        rollback.meshHolder->setVisualizePropertyMask( MeshVisualizePropertyType::PolygonOffsetFromCamera, rollback.polygonOffsetMask );
+    }
+}
+
+SubfeaturePickerWidgetLow::FindResult SubfeaturePickerWidgetLow::findSubfeature( const Object& subfeature )
+{
+    auto parent = subfeature.parent();
+    if ( !parent )
+        return {};
+
+    auto it = objects_.find( std::shared_ptr<Object>( std::shared_ptr<void>(), const_cast<Object *>( parent ) ) );
+    if ( it == objects_.end() )
+        return {};
+
+    auto subIt = std::find_if( it->second->subfeatures.begin(), it->second->subfeatures.end(), [&]( const SubfeatureData& sub ){ return sub.object.get() == &subfeature; } );
+    if ( subIt == it->second->subfeatures.end() )
+        return {}; // A child of the tracked object, but not one of the subfeatures.
+
+    return { &*it, &*subIt };
+}
+
+void SubfeaturePickerWidget::enable()
+{
+    if ( isEnabled_ )
+        return;
+
+    isEnabled_ = true;
+
+    connect( &getViewerInstance() );
+}
+
+void SubfeaturePickerWidget::disable()
+{
+    if ( !isEnabled_ )
+        return;
+
+    isEnabled_ = false;
+
+    disconnect();
+    underlyingPicker.removeTemporaryObjects();
+
+    targets.clear();
+
+    for ( TemporaryObject& elem : temporaryObjects_ )
+    {
+        if ( elem.object )
+            elem.object->detachFromParent();
+    }
+    temporaryObjects_.clear();
+}
+
+bool SubfeaturePickerWidget::drawGui( std::function<void()> preDrawGui )
+{
+    if ( !isEnabled_ )
+        return false;
+
+    if ( ImGui::GetIO().WantCaptureMouse )
+        mouseMovePick_ = {};
```

Comment:
same

Reply:
Doesn't work correctly without this one.